src/cls/queue/cls_queue_src.cc

```diff
@@ -0,0 +1,527 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#include "include/types.h"
+
+#include <errno.h>
+
+#include "objclass/objclass.h"
+#include "cls/queue/cls_queue_types.h"
+#include "cls/queue/cls_queue_ops.h"
+#include "cls/queue/cls_queue_const.h"
+#include "cls/queue/cls_queue_src.h"
+
+static string get_opaque_string_from_marker(const cls_queue_marker& marker) {
+  bufferlist bl, bl64;
+  encode(marker, bl);
+  bl.encode_base64(bl64);
+  string opaque_marker = bl64.to_str();
```

Comment:
the base64 encoding is a cool way to make the marker encodings printable, but i worry about the overhead of base64 and two temporary buffer allocations here. i think the cls_queue_marker::to_str() and from_str() functions are sufficient for these conversions

Reply:
Do you mean that instead of using base64 encoding to make the markers opaque, i can just use the cls_queue_marker::to_str() method to return opaque markers?

Reply:
yeah, do you buy that?

Reply:
I want to weigh in here :)
Opaque doesn't mean we need to obfuscate the data--it just means the client isn't allowed to interpret it.  We can use any format that works for us--so we shouldn't pay an extra cost for base64 encoding unless we need to construct a lexical token that stands in for the marker.